bundle/deploy/terraform/convert.go

```diff
@@ -275,5 +311,36 @@ func TerraformToBundle(state *tfjson.State, config *config.Root) error {
 		}
 	}
 
+	for _, src := range config.Resources.Jobs {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
+	for _, src := range config.Resources.Pipelines {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
+	for _, src := range config.Resources.Models {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
+	for _, src := range config.Resources.Experiments {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
+	for _, src := range config.Resources.ModelServingEndpoints {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
+	for _, src := range config.Resources.RegisteredModels {
+		if src.ModifiedStatus == "" && src.ID == "" {
+			src.ModifiedStatus = resources.ModifiedStatusCreated
+		}
+	}
```

Comment:
@ilia-db @andrewnester @pietern We keep adding new places in the code base where we need to do something specific for each resource type. That makes it harder and harder to correctly introduce a new resource type.

I wonder how we can best address that. One partial mitigation can be to use some reflection in tests to enforce that we cover all resource types, as seen in https://github.com/databricks/cli/blob/main/bundle/config/mutator/process_target_mode_test.go#L256-L257. For now, how about we just apply that pattern here as well?

Reply:
I'll update the tests (and add this additional check), they actually don't cover all resource types right now.

I think we can improve the situation if we move per-resource logic to the resource methods (like `TransformDevelopmentMode` or `UpdateModifiedStatus`) and then rely on interfaces in the functions like this one here. But maybe there are better options?

Reply:
Yeah, reorganizing the source structure would also help here. That way, as you add new resource types you don't need to edit the code in ten million places. Could be a followup though so we look at that holistically. Thoughts, @pietern?

Reply:
Yeah, that could work. But have to take into account the dynamic configuration tree by then as well. We're modifying the typed configuration here exclusively. For now, this is fine. Adding a new resource involves going to all call sites of other resources anyway, so having one extra is not the end of the world.